src/c/server_streaming_blocking_call.c

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ * Copyright 2015, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+
+#include <grpc_c/grpc_c.h>
+#include <grpc/grpc.h>
+#include <grpc/support/log.h>
+#include "server_streaming_blocking_call.h"
+#include <grpc_c/completion_queue.h>
+#include "alloc.h"
+#include "completion_queue.h"
+#include "tag.h"
+
+GRPC_client_reader *GRPC_server_streaming_blocking_call(GRPC_channel *channel,
+                                                        const GRPC_method rpc_method,
+                                                        GRPC_client_context *const context,
+                                                        const GRPC_message request) {
+  grpc_completion_queue *cq = GRPC_completion_queue_create();
+  grpc_call *call = grpc_channel_create_call(channel,
+                                             NULL,
+                                             GRPC_PROPAGATE_DEFAULTS,
+                                             cq,
+                                             rpc_method.name,
+                                             "",
+                                             context->deadline,
+                                             NULL);
+  context->call = call;
+  context->rpc_method = rpc_method;
+
+  grpc_call_op_set set = {
```

Comment:
I can't help but think this code would be simpler and more maintainable if we eliminated the call_op_set abstraction entirely and just wrote out the ops.

What are we gaining by having it?